pkg/splunk/enterprise/monitoringconsole.go

```diff
@@ -0,0 +1,347 @@
+	if err != nil {
+		return err
+	}
+
+	//configMapHash to trigger the configMap change in monitoring console deployment
+	configMapHash, err := getConfigMapHash(client, cr.GetNamespace())
+	if err != nil {
+		return err
+	}
+
+	deploymentMC, err := getMonitoringConsoleDeployment(cr, &spec, SplunkMonitoringConsole, 1, configMapHash, secretName)
+	if err != nil {
+		return err
+	}
+
+	_, err = splctrl.ApplyDeployment(client, deploymentMC)
+	if err != nil {
+		return err
+	}
+
+	return err
+}
+
+// GetMonitoringConsoleDeployment returns a Kubernetes Deployment object for Splunk Enterprise monitoring console instance.
+func getMonitoringConsoleDeployment(cr splcommon.MetaObject, spec *enterprisev1.CommonSplunkSpec, instanceType InstanceType, replicas int32, configMapHash string, secretName string) (*appsv1.Deployment, error) {
+	ports := splcommon.SortContainerPorts(getSplunkContainerPorts(SplunkMonitoringConsole))
+	annotations := splcommon.GetIstioAnnotations(ports)
+	var partOfIdentifier string
+	selectLabels := getSplunkLabels(cr.GetNamespace(), instanceType, partOfIdentifier) //using Namespace here so that with every CR the name should remain same Ex- splunk-<namespace>-monitoring-console
+	affinity := splcommon.AppendPodAntiAffinity(&spec.Affinity, cr.GetNamespace(), instanceType.ToString())
+	configMap := GetSplunkDefaultsName(cr.GetNamespace(), SplunkMonitoringConsole)
+
+	// start with same labels as selector; note that this object gets modified by splcommon.AppendParentMeta()
+	labels := make(map[string]string)
+	for k, v := range selectLabels {
+		labels[k] = v
+	}
+
+	//create deployment configuration
+	deploymentMC := &appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Deployment",
+			APIVersion: "apps/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      GetSplunkMonitoringConsoleDeploymentName(instanceType, cr.GetNamespace()),
+			Namespace: cr.GetNamespace(),
+		},
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: selectLabels,
+			},
+			Replicas: &replicas,
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels:      labels,
+					Annotations: annotations,
+				},
+				Spec: corev1.PodSpec{
+					Affinity:      affinity,
+					Tolerations:   spec.Tolerations,
+					SchedulerName: spec.SchedulerName,
+					Containers: []corev1.Container{
+						{
+							Image:           spec.Image,
+							ImagePullPolicy: corev1.PullPolicy(spec.ImagePullPolicy),
+							Name:            "splunk",
+							Ports:           ports,
+							EnvFrom: []corev1.EnvFromSource{
+								{
+									ConfigMapRef: &corev1.ConfigMapEnvSource{
+										LocalObjectReference: corev1.LocalObjectReference{
+											Name: configMap, //monitoring console env variables configMap
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	if configMapHash == "" {
```

Comment:
This is just an extra check. Already taking care of empty value in configMapHash, Since we will always create an empty configMap so this value will never be null

Reply:
Can the extra check be at the beginning of the function, even before deployment config is created?